index.js

```diff
@@ -0,0 +1,434 @@
+        else
+            next = ARGV[i]
+        end
+        if i == #ARGV-1 then
+            break;
+        end
+        if not path[next] then
+            path[next] = {};
+        end
+        path = path[next];
+    end
+    path[next] = cjson.decode(ARGV[#ARGV]);
+    -- convert and set the value
+    return redis.call('SET', KEYS[1], cjson.encode(data));
+`;
+
+// This lua script deletes a nested property atomically
+// Usage: EVALSHA(SHA, 1, key, property, [property...]);
+// e.g. Delete obj.a.b.c -> EVALSHA(SHA, 1, 'obj', 'a', 'b', 'c');
+const deleteScript = `
+    -- get the value of key
+    local data = redis.call('GET', KEYS[1]);
+    if data then
+        data = cjson.decode(data);
+    end
+    -- parse path
+    local path = data;
+    local next = tonumber(ARGV[1]);
+    for i = 2, #ARGV do
+        if not path then
+            return 0;
+        end
+        if next then
+            path = path[next+1];
+        else
+            path = path[ARGV[i-1]];
+        end
+        next = tonumber(ARGV[i]);
+    end
+    -- delete the property
+    if next and path[next+1] then
+        table.remove(path, next+1);
+    elseif path[ARGV[#ARGV]] then
+        path[ARGV[#ARGV]] = nil;
+    else
+    -- return if try to delete non-existent value
+        return 0
+    end
+    -- convert and set the value
+    return redis.call('SET', KEYS[1], cjson.encode(data));
+`;
+
+function stringify(value) {
+    let hasCircular;
+    let result = safeJSONStringify(value, null, null, function (k, v) { hasCircular = true; });
+    return { json: result, circular: hasCircular };
+}
+
+function addPrefix(prefix, scope, key) {
+    if (prefix) {
+        scope = prefix + ':' + scope;
+    }
+    return scope + ':' + key;
+}
+
+function removePrefix(prefix, scope, key) {
+    if (prefix) {
+        key = key.substring((prefix + ':').length);
+    }
+    return key.substring((scope + ':').length);
+}
+
+function scan(client, pattern, cursor = 0) {
+    return new Promise((resolve, reject) => {
+        client.SCAN(cursor, 'MATCH', pattern, 'COUNT', 1000, (err, results) => {
+            if (err) {
+                return reject(err);
+            } else {
+                const cursor = results[0];
+                const elements = results[1];
+                if (cursor === "0") {
+                    //the iteration finished
+                    resolve(elements);
+                } else {
+                    scan(client, pattern, cursor).then(result => {
+                        resolve(elements.concat(result));
+                    });
+                }
+            }
+        });
+    });
+}
+
+function Redis(config) {
+    this.host = config.host || '127.0.0.1';
+    this.port = config.port || 6379;
+    this.prefix = config.prefix;
+    this.options = {
+        db: config.db || 0,
+        password: config.password,
+        tls: config.tls,
+        //Do not try to reconnect, throw an error
+        retry_strategy: () => undefined
```

Comment:
Any reason not to reconnect? If this is set to undefined and the connection is lost to the db, then no attempts to reconnect will happen until the plugin is re-initialized which is at node-red container startup. I would suggest removing this line of code unless there is a reason to not reconnect. 

Here is more context regarding this issue,

https://github.com/node-red/node-red-context-redis/issues/2